gateway/ln-gateway/src/gatewayd/actor.rs

```diff
@@ -57,7 +75,114 @@ impl GatewayActor {
             }
         });
 
-        Ok(Self { client })
+        let actor = Self {
+            client,
+            lnrpc,
+            task_group,
+        };
+
+        actor.subscribe_htlcs().await?;
+
+        Ok(actor)
+    }
+
+    async fn subscribe_htlcs(&self) -> Result<()> {
+        let actor = self.to_owned();
+        let lnrpc_copy = self.lnrpc.to_owned();
+        let short_channel_id = self.client.config().mint_channel_id;
+        let mut tg = self.task_group.clone();
+
+        let mut stream = lnrpc_copy
+            .subscribe_intercept_htlcs(SubscribeInterceptHtlcsRequest { short_channel_id })
+            .await?;
+
+        tg.spawn(
+            "Subscribe to intercepted HTLCs in stream",
+            move |subscription| async move {
+                while let Some(SubscribeInterceptHtlcsResponse {
+                    payment_hash,
+                    outgoing_amount_msat,
+                    intercepted_htlc_id,
+                    ..
+                }) = match stream.message().await {
+                    Ok(Some(msg)) => Some(msg),
+                    Ok(None) => {
+                        warn!("HTLC stream closed by service");
+                        None
+                    }
+                    Err(e) => {
+                        error!("HTLC stream closed with error: {:?}", e);
+                        None
+                    }
+                } {
```

Comment:
IMHO, the struct unpacking and matching in one big block looks sort of weird. Maybe it's considered idiomatic rust, maybe not; I don't know. But I'd prefer to structure this a bit different (see next comment)

Reply:
I'm certainly not against unpacking, that's a pretty cool feature and I'm quite sure there are no performance costs either. 

I think we'll see what this loop ends up with, when `await`ing the `stream.message()` and the shutdown handler of the task group